internal/fields/validate.go

```diff
@@ -154,6 +159,11 @@ func isFieldFamilyMatching(family, key string) bool {
 	return key == family || strings.HasPrefix(key, family+".")
 }
 
+func isFieldTypeFlattened(key string, fieldDefinitions []FieldDefinition) bool {
+	definition := findElementDefinitionRoot(key, fieldDefinitions)
```

Comment:
nit: Actually I was thinking to call it the other way round, sorry for not being clear. The `findElementDefinition` seems to be more generic than `findElementDefinition(WithRoot)`.

Reply:
Done.